src/sumo_backend_riak.erl

```diff
@@ -81,24 +85,63 @@ init(Options) ->
-    proplists:get_value(bucket, Options, <<"sumo_test">>)),
-  Index = iolist_to_binary(
-    proplists:get_value(index, Options, <<"sumo_test_index">>)),
-  %% Place Riak connection
-  {ok, Conn} = riakc_pb_socket:start_link(Host, Port, Opts),
+  %% Create Riak connection pool
+  F = fun(_E, Acc) ->
+        {ok, Conn} = riakc_pb_socket:start_link(Host, Port, Opts),
+        [Conn | Acc]
+      end,
+  ConnPool = lists:foldl(F, [], lists:seq(1, PoolSize)),
   %% Initial state
-  {ok, #state{conn = Conn, bucket = {BucketType, Bucket}, index = Index}}.
-
+  {ok, #state{conn_args = {Host, Port, Opts}, conn_pool = ConnPool}}.
+
+%% @todo: These are workarounds, a real connection pool needs to be added.
+%% Workaround 1: when the store calls 'get_connection', a new Riak connection
+%% is returned - one to one model (between connection and store).
+%% Workaround 2: a simple list (LIFO) was added to hold a set of connections
+%% that are created in the 'init' function (see code above). When the store
+%% needs a connection, must call 'checkout_conn' to get the Pid, and when
+%% it finish, must call 'checkin_conn'. These operations have some problems,
+%% for instance, the don't consider an overflow, so the amount of new
+%% connection are not watched, so in case of high concurrency, exist the
+%% risk of have so many connection hitting the DB and causing contention.
 -spec handle_call(term(), term(), state()) -> {reply, term(), state()}.
-handle_call(get_connection, _From, State = #state{conn = Conn}) ->
+handle_call(get_connection,
+            _From,
+            State = #state{conn_args = {Host, Port, Opts}}) ->
+  {ok, Conn} = riakc_pb_socket:start_link(Host, Port, Opts),
   {reply, Conn, State};
-handle_call(get_state, _From, State) ->
-  {reply, State, State}.
+handle_call({checkin_conn, Conn},
+            _From,
+            State = #state{conn_args = {Host, Port, Opts},
+                           conn_pool = ConnPool}) ->
+  NewConn = case is_process_alive(Conn) of
+              true ->
+                Conn;
+              false ->
+                {ok, Conn0} = riakc_pb_socket:start_link(Host, Port, Opts),
+                Conn0
+            end,
+  {reply, ok, State#state{conn_pool = [NewConn | ConnPool]}};
+handle_call(checkout_conn,
+            _From,
+            State = #state{conn_args = {Host, Port, Opts}, conn_pool = []}) ->
+  {ok, Conn} = riakc_pb_socket:start_link(Host, Port, Opts),
+  {reply, Conn, State#state{conn_pool = [Conn]}};
+handle_call(checkout_conn,
+            _From,
+            State = #state{conn_args = {Host, Port, Opts},
+                           conn_pool = ConnPool}) ->
+  [Conn | _T] = ConnPool,
+  NewConn = case is_process_alive(Conn) of
+              true ->
+                Conn;
+              false ->
+                {ok, Conn0} = riakc_pb_socket:start_link(Host, Port, Opts),
+                Conn0
+            end,
+  {reply, NewConn, State#state{conn_pool = (ConnPool -- [Conn])}}.
```

Comment:
This is __extremely__ inneficient and actually doesn't work really well.

First, the ``--`` should be avoided and you have _T which is the same list you get with ConnPool -- [Conn], second, you trust ``is_process_alive`` too much, when that's not how it should be used, it could die immediately after you make the call.